src/vmm/src/utilities/test_utils/mod.rs

```diff
@@ -5,15 +5,40 @@
 use std::sync::{Arc, Mutex};
 
 use utils::tempdir::TempDir;
+use vm_memory::GuestAddress;
 
 use crate::builder::build_microvm_for_boot;
 use crate::resources::VmResources;
 use crate::seccomp_filters::get_empty_filters;
 use crate::utilities::mock_resources::{MockBootSourceConfig, MockVmConfig, MockVmResources};
 use crate::vmm_config::boot_source::BootSourceConfig;
 use crate::vmm_config::instance_info::InstanceInfo;
+use crate::vstate::memory::{GuestMemoryExtension, GuestMemoryMmap};
 use crate::{EventManager, Vmm};
 
+/// Creates a [`GuestMemoryMmap`] with a single region of the given size starting at guest
+/// physical address 0 and without dirty tracking.
+pub fn single_region_mem(region_size: usize) -> GuestMemoryMmap {
+    single_region_mem_at(0, region_size)
+}
+
+/// Creates a [`GuestMemoryMmap`] with a single region of the given size starting at the given
+/// guest physical address `at` and without dirty tracking.
+pub fn single_region_mem_at(at: u64, size: usize) -> GuestMemoryMmap {
+    multi_region_mem(&[(GuestAddress(at), size)])
+}
+
+/// Creates a [`GuestMemoryMmap`] with multiple regions and without dirty page tracking.
+pub fn multi_region_mem(regions: &[(GuestAddress, usize)]) -> GuestMemoryMmap {
+    GuestMemoryMmap::from_raw_regions(regions, false).expect("Cannot initialize memory")
+}
+
+/// Creates a [`GuestMemoryMmap`] of the given size with the contained regions laid out in
+/// accordance with the requirements of the architecture on which the tests are being run.
+pub fn arch_mem(mem_size_bytes: usize) -> GuestMemoryMmap {
```

Comment:
nit: Maybe `default_mem` instead of `arch_mem`. `arch_mem` seems too architecture specific.

Reply:
Mh, it kinda is architecture specific, because its return value is the default memory configuration for the architecture on which we're runnning. So am not a big fan of just `default_mem` :/. Maybe someone has an idea how to combine them